test/unittest/dds/participant/ParticipantTests.cpp

```diff
@@ -1992,6 +2116,87 @@ TEST(ParticipantTests, GetSubscriberProfileQos)
     ASSERT_EQ(DomainParticipantFactory::get_instance()->delete_participant(participant), RETCODE_OK);
 }
 
+TEST(ParticipantTests, GetSubscriberQosFromXml)
+{
+    const std::string xml_filename("test_xml_profile.xml");
+    const std::string profile_name("test_subscriber_profile");
+
+    std::string complete_xml = testing::load_file(xml_filename);
+
```

Comment:
Maybe is good to create the participants disable in this tests using the `DomainParticipantFactoryQos`.